src/Knp/Component/Pager/Event/Subscriber/Paginate/Doctrine/DBALQueryBuilderSubscriber.php

```diff
@@ -22,17 +22,17 @@ public function items(ItemsEvent $event): void
             $qb = clone $target;
             
             //reset count orderBy since it can break query and slow it down
-            $qb
-                ->resetQueryPart('orderBy')
-            ;
-            
+            if (method_exists($qb, 'resetOrderBy')) {
+                $qb->resetOrderBy();
+            } else {
+                $qb->resetQueryParts(['orderBy']);
+            }
+
             // get the query
             $sql = $qb->getSQL();
-            
+
             $qb
-                ->resetQueryParts()
                 ->select('count(*) as cnt')
-                ->from('(' . $sql . ')', 'dbal_count_tbl')
             ;

```

Comment:
`resetQueryParts()` has been removed in DBALv4.

Removing `->from()` makes the test pipeline green.
However, I'm not sure if this could cause any side effects.

Without this change `$qb->getSQL()` would return `SELECT count(*) as cnt FROM Article a, (SELECT * FROM Article a) dbal_count_tbl` which would lead to the result `16` instead of `4`

In `master`, `$qb->getSQL()` returns `SELECT count(*) as cnt FROM (SELECT * FROM Article a) dbal_count_tbl`.